File: Data Structures & Algorithms/is-subsequence/submission-3.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(t) < len(s):
            return False

        p1 = p2 = 0

        while p2 < len(t) and p1 < len(s):
            if t[p2] == s[p1]:
                p1 += 1
            p2 += 1
        
        print(p1, p2)
        return (p1 == len(s))

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
#include <unordered_map>
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_map<int, int> hashMap;

        for(int i = 0; i < nums.size(); i++) {
            if (hashMap.contains(nums[i])) {
                return true;
            }
            hashMap[nums[i]] = 1;
        }
        return false;
    }
};